deps/thpool/thpool.h

```diff
@@ -273,6 +293,72 @@ void redisearch_thpool_destroy(redisearch_threadpool);
  */
 size_t redisearch_thpool_num_threads_working(redisearch_threadpool);
 
+/* ============ COLLECT THREADS DATA AND LOG API ============ */
```

Comment:
consider moving log api + implementation to different h+c files as they are not really part of the thpool

Reply:
Good idea and i'll try